src/transactions/feed/TransactionFeedV2.tsx

```diff
@@ -27,8 +31,131 @@ type PaginatedData = {
 const POLL_INTERVAL = 10000 // 10 secs
 const FIRST_PAGE_TIMESTAMP = 0
 
-function getAllowedNetworkIdsForTransfers() {
-  return getMultichainFeatures().showTransfers.join(',').split(',') as NetworkId[]
+function getAllowedNetworksForTransfers() {
+  return getMultichainFeatures().showTransfers
+}
+
+/**
+ * Join allowed networks into a string to help react memoization.
+ * N.B: This fetch-time filtering does not suffice to prevent non-Celo TXs from appearing
+ * on the home feed, since they get cached in Redux -- this is just a network optimization.
+ */
+function useAllowedNetworksForTransfers() {
+  const allowedNetworks = getAllowedNetworksForTransfers().join(',')
+  return useMemo(() => allowedNetworks.split(',') as NetworkId[], [allowedNetworks])
+}
+
+/**
+ * Join supported networks for approval into a string to help react memoization.
+ * N.B: This fetch-time filtering does not suffice to prevent non-Celo TXs from appearing
+ * on the home feed, since they get cached in Redux -- this is just a network optimization.
+ */
+function useSupportedNetworksForApproval() {
+  const supportedNetworks = getSupportedNetworkIdsForApprovalTxsInHomefeed().join(',')
+  return useMemo(() => supportedNetworks.split(',') as NetworkId[], [supportedNetworks])
+}
+
+/**
+ * This function uses the same deduplication approach as "deduplicateTransactions" function from
+ * queryHelper but only for a single flattened array instead of two.
+ * Also, the queryHelper is going to be removed once we fully migrate to TransactionFeedV2,
+ * so this function would have needed to be moved from queryHelper anyway.
+ */
+function deduplicateTransactions(transactions: TokenTransaction[]): TokenTransaction[] {
+  const transactionMap: { [txHash: string]: TokenTransaction } = {}
+
+  for (const tx of transactions) {
+    transactionMap[tx.transactionHash] = tx
+  }
+
+  return Object.values(transactionMap)
+}
+
+/**
+ * If the timestamps are the same, most likely one of the transactions is an approval.
+ * On the feed we want to show the approval first.
+ */
+function sortTransactions(transactions: TokenTransaction[]): TokenTransaction[] {
+  return transactions.sort((a, b) => {
+    const diff = b.timestamp - a.timestamp
+
+    if (diff === 0) {
+      return a.__typename === 'TokenApproval' ? 1 : b.__typename === 'TokenApproval' ? -1 : 0
```

Comment:
Trying to use `tx.type` instead of the GraphQL provided `__typename` since we're gonna remove GraphQL anyway.
```suggestion
      return a.type === TokenTransactionTypeV2.Approval ? 1 : b.type === TokenTransactionTypeV2.Approval ? -1 : 0
```